### PR TITLE
Fixed AM PM for 12 noon

### PR DESCRIPTION
**Pull request by Moltenkore**
At the moment 12 noon is parsed as 12 AM which is incorrect.

### CODE REVIEW

**Comment by loilo** [comment]
I'm not familiar with what's "correct" (living in a country with a 24 hour system), but this is [how PHP handles it](http://sandbox.onlinephpfunctions.com/code/7c1c20ff754203c97a0e482ca4fb6b32642c1831), and this library aims to yield the same results as PHP's `date()`.

Sorry — thanks for putting thought and work into this!

**Comment by Moltenkore** [comment]
If you test that with the link provided it gives a result of ``12pm`` where as the Javascript lib gives a result of ``12am``. So it is mimicking the correct behaviour by returning 12 noon.

```
php > echo date('Ha', strtotime('2021-03-15 12:00:00.000'));
12pm

// Before
let hours = 12; console.log( hours <= 12 ? 'am' : 'pm' );
am

// After
let hours = 12; console.log( hours <= 12 ? hours == 12 ? 'pm' : 'am' : 'pm' );
pm
```

**Comment by loilo** [comment]
You're absolutely right. Sorry, I am stupid.
I just read the diff the wrong way around. 🤦‍♂️

File: src/date.js [hunk 1]
```diff
@@ -144,8 +144,8 @@ function formatDateBase(formatterString, date, utc) {
     o: () => calcISOWeek().year,
     Y: () => year,
     y: () => String(year).substr(2),
-    a: () => (hours <= 12 ? 'am' : 'pm'),
-    A: () => (hours <= 12 ? 'AM' : 'PM'),
+    a: () => (hours <= 12 ? hours == 12 ? 'pm' : 'am' : 'pm'),
+    A: () => (hours <= 12 ? hours == 12 ? 'PM' : 'AM' : 'PM'),
```
Comment (loilo) [inline src/date.js:148]: The suggested Change is a bit convulted, this short version should do it, shouldn't it?

```suggestion
    a: () => (hours < 12 ? 'am' : 'pm'),
    A: () => (hours < 12 ? 'AM' : 'PM'),
```

**Review by loilo — CHANGES_REQUESTED** [review]

**Comment by Moltenkore** [comment]
You're right, that's more elegant now.

**Comment by loilo** [comment]
Alright, thanks a lot for your help!
3.0.1 is out with the fix. 🎉

**Pull request merged**